azure-pipelines.yml

```diff
@@ -0,0 +1,164 @@
+
+        - task: PowerShell@2
+          displayName: 'Pre-release versioning - CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $changelogContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -Raw
+
+              Write-Host " (i) Current version: $env.currentVersion"
+              Write-Host " (i) Original CHANGELOG.md content: $changelogContent"
+
+              [string] $newChangelogContent = $changelogContent -replace $env:currentVersion, "$env:currentVersion-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated CHANGELOG.md content: $newChangelogContent"
+
+              $newChangelogContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md"
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: ReleaseProd
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/master'))
+    jobs: 
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+
+        - task: PowerShell@2
+          displayName: 'Get version from pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+        - task: GitHubRelease@1
+          displayName: 'Tag a release to GitHub'
+          inputs:
+            gitHubConnection: 'github.com_Brett09'
+            repositoryName: 'builttoroam/flutter_plugins'
+            action: 'create'
+            target: 'master'
+            tagSource: 'userSpecifiedTag'
+            tag: v$(currentVersion)
+            title: v$(currentVersion)
```

Comment:
We should pull out the text from version line in the change log. In the past this has just been the date of the release but I think going forward this should be a single line summary of the change eg Added Timezone Support

Reply:
What if there are multiple items (eg. multiple new feature added)? As an example you can see the v3.1.0 changelog in `develop` branch, there are a lot of changes.

Maybe from now on we should have a habbit of having a summary in the first line of each version in the changelog?

Reply:
correct that's what I was thinking - just have it on same line as the version number eg a title

Reply:
We usuall have a release date on the same line of the version (i.e. for version 3.0.0 we have: `3.0.0 21 Janurary 2020`)

Do you still want to put the date and put a summary after? or forget about the date?